vayesta/core/scmf/scmf.py

```diff
@@ -85,7 +85,8 @@ def kernel(self, *args, **kwargs):
             self.log.info("%s==============", len(self.name)*"=")
 
             if self.iteration > 1:
-                self.emb.reset()
+                # Don't reset inactive clusters for external corrections
+                self.emb.reset(reset_inactive=False)
```

Comment:
I would suggest two ways to combine `scmf` with external corrections:
* add a `callback` to `scmf`, which is called each iteration before the kernel,
and can be used to update the FCI clusters.
* support the FCI fragments more natively, for example introduce an "auxiliary" flag for fragments, with the following effects:
   * Both auxiliary and non-auxiliary fragments are solved, when calling `emb.kernel()`
   * Auxiliary fragments are solved before any non-auxiliary fragments
   * Auxiliary fragments do not contribute to expectation values
   
   This would then get around the whole "setting active/inactive" script level code

Reply:
I guess the second option would likely be cleaner (especially for any end-user). I'm trying to think of drawbacks (we are getting quite a few fragment flags now, but perhaps thats ok...). I suggest we go for that one, unless you can think of a reason not to...?